src/Bounded.ts

```diff
@@ -17,11 +17,105 @@ import { Ord, ordNumber } from './Ord'
  * @category type classes
  * @since 2.0.0
  */
-export interface Bounded<A> extends Ord<A> {
+export interface Bounded<A> extends Ord.Ord<A> {
   readonly top: A
   readonly bottom: A
 }
 
+// -------------------------------------------------------------------------------------
+// deconstructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category deconstructors
+ * @since 2.0.0
+ */
+export const top = <T>(b: Bounded<T>) => b.top
+
+/**
+ * @category deconstructors
+ * @since 2.0.0
+ */
+export const bottom = <T>(b: Bounded<T>) => b.bottom
+
+/**
+ * Returns the tuple [bottom, top].
+ *
+ * @category deconstructors
+ * @since 2.0.0
+ */
+export const toTuple = <T>(bound: Bounded<T>): [T, T] =>
+    [bound.bottom, bound.top]
+
+// -------------------------------------------------------------------------------------
+// guards
+// -------------------------------------------------------------------------------------
+
+/**
+ * Test that top >= bottom
+ *
+ * @category guards
+ * @since 2.0.0
+ */
+export const isValid = <T>(bound: Bounded<T>) =>
+    Ord.geq(bound)(bound.bottom, bound.top)
+
+// -------------------------------------------------------------------------------------
+// constructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * Returns an instance of Bounded from a range of values.
+ * Returns none if bottom > top and some if top >= bottom.
+ *
+ * @category constructors
+ * @since 2.0.0
+ */
+export const fromRange = <T>(ord: Ord.Ord<T>) => (b: T) => (t: T): Bounded<T> =>
```

Comment:
The comment says that you return want to return `O.Option<Bounded<T>>` but it is missing in your signature and function logic.